Allow for parameters to be sent to Base class

Kartstig opened the pull request:
I would like to propose the following changes so the BAC0 instance can be configured.

- `firmwareRevision` - Rather than setting the Python version, I'd like to use this to indicate the version of the BAC0 software I'm writing
- `modelName` - This will be used to describe the particular hardware / firmware that is running my BAC0 code
- `vendorId` - I will be updating this after applying / registering for a specific vendorId with BAC0
- `vendorName` - This will be used to identify my organization

If you would like me to format the code in any way or apply the same changes to `Complete.py`, I would be happy to.

Merci!

ChristianTremblay commented:
Thanks for this. I never used `**param` what does it do ?

Regarding format, I "black" everything :-)

https://github.com/python/black

Kartstig commented:
Oh! `**my_variable` is a convenient way to destructure keyword arguments into a `dict`

Here's a link to a description in [Python docs](https://docs.python.org/dev/tutorial/controlflow.html#keyword-arguments)

Here's an example of where I've found it quite useful:
```python
class User:
    def __init__(self, email, password=None, **kwargs):
        timestamp = datetime.utcnow()
        self.email              = email
        self.password           = (self.pass_hash(password)) if password else None
        self.first_name         = kwargs.get('first_name', None)
        self.last_name          = kwargs.get('last_name', None)
        self.phone_number       = self.parse_phone(kwargs.get('phone_number', None))
        self.role               = kwargs.get('role', 'user')
        self.created_at         = timestamp
        self.updated_at         = timestamp
        self.validation_token   = b64encode(os.urandom(32)).decode('utf-8')
        self.last_login         = kwargs.get('last_login', None)
        self.valid_email        = kwargs.get('valid_email', False)
        self.is_subscribed      = kwargs.get('is_subscribed', True)
        self.active             = kwargs.get('active', True)
```

you can use this like:
```python
User(email, password, first_name="herman")
```
`kwargs` becomes a dictionary and
```python
kwargs.get('first_name')
=> "herman"
kwargs.get('last_name')
=> None
```

Kartstig commented:
This should be ready for review.

I added the passing of parameters to `Complete.py` so `BAC0.lite()` and `BAC0.connet()` exhibit the same behavior.

Everything was formatted with `black`

I added a couple basic tests around checking that the parameters passed indeed get set onto the instance. Let me know if there are any other changes to be made!
